Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/sdk/storage/ADLSStorageAccount.java

```diff
@@ -35,13 +34,15 @@
     private final IClusterDetail clusterDetail;
     private final ClusterIdentity clusterIdentity;
     private ADLSCertificateInfo certificateInfo;
+    private final String defaultStorageSchema;
 
-    public ADLSStorageAccount(IClusterDetail clusterDetail, String name, boolean isDefault, String defaultRootPath, ClusterIdentity clusterIdentity) {
+    public ADLSStorageAccount(IClusterDetail clusterDetail, String name, boolean isDefault, String defaultRootPath, ClusterIdentity clusterIdentity, String storageSchema) {
```

Comment:
Could add a new constructor with root folder URI as a parameter like this?

```suggestion
    public ADLSStorageAccount(IClusterDetail clusterDetail, boolean isDefault, URI rootUri, ClusterIdentity clusterIdentity) {
```